wait/host.go

```diff
@@ -0,0 +1,152 @@
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"reflect"
+)
+
+type HostAwaitility struct {
+	*SingleAwaitilityImpl
+}
+
+func NewHostAwaitility(a *Awaitility) *HostAwaitility {
+	return &HostAwaitility{
+		SingleAwaitilityImpl: &SingleAwaitilityImpl{
+			T:               a.T,
+			Client:          a.Client,
+			Ns:              a.HostNs,
+			OtherOperatorNs: a.MemberNs,
+		}}
+}
+
+func (a *HostAwaitility) WaitForMasterUserRecord(name string) error {
+	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		mur := &toolchainv1alpha1.MasterUserRecord{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, mur); err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("waiting for availability of MasterUserRecord '%s'", name)
+				return false, nil
+			}
+			return false, err
+		}
+		a.T.Logf("found MasterUserAccount '%s'", name)
+		return true, nil
+	})
+}
+
+func (a *HostAwaitility) GetMasterUserRecord(name string) *toolchainv1alpha1.MasterUserRecord {
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, mur)
+	require.NoError(a.T, err)
+	return mur
+}
+
+func (a *HostAwaitility) WaitForMurConditions(name string, waitCond ...MurWaitCondition) error {
+	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		mur := &toolchainv1alpha1.MasterUserRecord{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, mur); err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("waiting for availability of MasterUserRecord '%s'", name)
+				return false, nil
+			}
+			return false, err
+		}
+		for _, isMatched := range waitCond {
+			if !isMatched(a, mur) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+}
+
+type MurWaitCondition func(a *HostAwaitility, mur *toolchainv1alpha1.MasterUserRecord) bool
+
+func UntilHasStatusCondition(conditions ...toolchainv1alpha1.Condition) MurWaitCondition {
+	return func(a *HostAwaitility, mur *toolchainv1alpha1.MasterUserRecord) bool {
+		if test.ConditionsMatch(mur.Status.Conditions, conditions...) {
+			a.T.Logf("status conditions match in MasterUserRecord '%s`", mur.Name)
+			return true
+		}
+		a.T.Logf("waiting for correct status condition of MasterUserRecord '%s`", mur.Name)
+		return false
+	}
+}
+
+func UntilHasUserAccountStatus(expUaStatuses ...toolchainv1alpha1.UserAccountStatusEmbedded) MurWaitCondition {
+	return func(a *HostAwaitility, mur *toolchainv1alpha1.MasterUserRecord) bool {
+		if len(mur.Status.UserAccounts) != len(expUaStatuses) {
+			a.T.Logf("waiting for corrent number of UserAccount statuses in MasterUserRecord '%s`", mur.Name)
```

Comment:
waiting for corrent number of UserAccount ~~corrent~~ current

Reply:
fixed 3587446